# **Processing & Analyzing the Data**

First we take a look of the structure of the dataset with the objective of seek which variable we have to analyze and create:

In [16]:
SELECT TOP 10
    *
FROM [202103-divvy-tripdata]

(10 rows affected)

Total execution time: 00:00:00.028

ride_id rideable_type started_at ended_at start_station_name start_station_id end_station_name end_station_id start_lat start_lng end_lat end_lng member_casual 0000465CF790D824 electric_bike 2021-03-08 18:08:23.0000000 2021-03-08 19:34:29.0000000 NULL NULL NULL NULL 4192 -8772 4193 -8767 casual 00004F0F77E12E18 classic_bike 2021-03-24 17:08:00.0000000 2021-03-24 18:20:07.0000000 Calumet Ave & 18th St 13102 Michigan Ave & 18th St 13150 41857612 -87619408 41857812 -8762455 casual 000081CDC629355A classic_bike 2021-03-08 17:48:21.0000000 2021-03-08 18:01:40.0000000 Ashland Ave & Grand Ave 13434 Ashland Ave & Grand Ave 13434 41891072 -87666608 41891072 -87666608 member 0000D6609083F8C8 docked_bike 2021-03-13 00:48:47.0000000 2021-03-13 01:51:12.0000000 Rush St & Hubbard St KA1503000044 Rush St & Hubbard St KA1503000044 41890172 -87626184 41890172 -87626184 casual 0000E746AFFDE10D classic_bike 2021-03-07 15:56:11.0000000 2021-03-07 16:14:26.0000000 Leavitt St & Division St 658 Clinton St & Lake St 13021 41902996 -87683824 41885636 -87641824 member 00013A098E40CBAD classic_bike 2021-03-09 16:59:32.0000000 2021-03-09 17:11:37.0000000 Theater on the Lake TA1308000001 Broadway & Barry Ave 13137 41926276 -87630832 4193758270717952 -876440958533632 casual 00013FFF86FF03DA classic_bike 2021-03-08 16:35:49.0000000 2021-03-08 17:16:21.0000000 Lake Shore Dr & Monroe St 13300 Lake Shore Dr & Monroe St 13300 41880960 -87616744 41880960 -87616744 casual 0001BC6649540A9E electric_bike 2021-03-09 09:00:54.0000000 2021-03-09 09:06:04.0000000 Ritchie Ct & Banks St KA1504000134 Clybourn Ave & Division St TA1307000115 41906936504909820 -8762620731457536 41904617222569980 -8764043976245248 member 0001DA2D84B47067 classic_bike 2021-03-11 17:10:19.0000000 2021-03-11 17:42:06.0000000 St. Clair St & Erie St 13016 Clark St & Grace St TA1307000127 4189434580828160 -8762279818428416 4195078 -87659168 member 00020B17897E5C2E classic_bike 2021-03-09 06:58:19.0000000 2021-03-09 07:18:56.0000000 Clark St & Grace St TA1307000127 Sheffield Ave & Waveland Ave TA1307000126 4195078 -87659168 41949400 -87654528 member

So, after having analyzed our dataset structure, we make some calculations in order to get some inights about differences found between types of users (**member\_casual**). I am going to include some aggregate functions like **COUNT, SUM** and **AVG:**

In [1]:
--Numero de viajes, Promedio de viajes y tiempo total de viajes, por tipo de usuario (member or casual)

SELECT  
	member_casual,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		member_casual, 
		DATEDIFF(SECOND,started_at, ended_at) AS ride_length 
	FROM [202103-divvy-tripdata]
) AS temp_table
GROUP BY member_casual

(2 rows affected)

Total execution time: 00:00:00.044

member_casual number_of_trips average_ride total_time_trips member 144463 838 121088499 casual 84033 2289 192395561

The _casual riders_ have less **number\_of\_trips** but a longer **average\_ride** time (in minutes) than  _member riders_. Because of that, the sum of **total\_time\_trips** is greater in _casual riders_. We have to explore why or how this happen.

Then we search a relationship between the **day\_of\_week** the user started the trip, by type of user (**member\_casual**) in order to indentify some differences. But first, I look up for the mode of **date\_of\_week** in general terms:

In [15]:
SELECT  
	day_of_week,
	COUNT(day_of_week) AS number_trips
FROM (
	SELECT	
		member_casual, 
		rideable_type,
		DATEPART(DW, started_at) AS day_of_week  
	FROM [202103-divvy-tripdata]
) AS temp_table1
GROUP BY day_of_week
ORDER BY number_trips DESC

(7 rows affected)

Total execution time: 00:00:00.059

day_of_week number_trips 7 45252 1 35627 2 34825 3 34000 4 31669 6 25657 5 21466

As you can see, there is an important preference for 7 (Sunday), following by 1 (Monday) in which **day\_of\_week** the user starts its ride (_Remember this: Monday=1 and Sunday=7)._
Now we try to analyze the preference of **day\_of\_week** separated by type of user (**member\_casual**) in order to keep trying find differences. Also we aggregate functions like **COUNT, AVG** and **SUM,** and group them by type of user(**member\_casual**) and **day\_of\_week** in order to get further detail about our analysis.

In [4]:

SELECT  
	member_casual,
	day_of_week,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		member_casual, 
		rideable_type,
		DATEDIFF(SECOND,started_at, ended_at) AS ride_length,  
		DATEPART(DW, started_at) AS day_of_week	
	FROM [202103-divvy-tripdata]
) AS temp_table
WHERE member_casual='casual'
GROUP BY member_casual, day_of_week
ORDER BY member_casual, number_of_trips DESC
------------------------------------------------------------------

SELECT  
	member_casual,
	day_of_week,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		member_casual, 
		rideable_type,
		DATEDIFF(SECOND,started_at, ended_at) AS ride_length,  
		DATEPART(DW, started_at) AS day_of_week	
	FROM [202103-divvy-tripdata]
) AS temp_table
WHERE member_casual='member'
GROUP BY member_casual, day_of_week
ORDER BY member_casual, number_of_trips DESC

(7 rows affected)

(7 rows affected)

Total execution time: 00:00:00.120

member_casual day_of_week number_of_trips average_ride total_time_trips casual 7 22090 2537 56063610 casual 1 17342 2481 43041420 casual 2 11990 2648 31752832 casual 3 10463 2160 22608134 casual 4 8883 1724 15316765 casual 6 7777 1751 13623522 casual 5 5488 1820 9989278

member_casual day_of_week number_of_trips average_ride total_time_trips member 3 23537 808 19037584 member 7 23162 952 22060680 member 2 22835 847 19357166 member 4 22786 775 17666738 member 1 18285 969 17721068 member 6 17880 765 13678611 member 5 15978 723 11566652

From the tables shown above, we can conclude the following:

- There is differences of which **day\_of\_week** each type of user prefer take the ride. The **number\_of\_trips** column shows us that _casual riders_ prefer Sunday followed by Monday, and _member riders_ prefer Wednesday followed by Sunday.
- Also, wee can see which **day\_of\_week** the **average\_ride** time (in minutes) is longer. In the case of _casual riders_, the **average\_ride** time is longer in Tuesday (44 minutes), Sunday (42) and Monday (41).
- For _member riders_, the longer **average\_ride** time are shorter than _casual riders:_ Monday (16 minutes), Sunday (15 ) and Tuesday (14). The average times of _member riders_ are more atomized than _casual riders._
- Notice that there is not much correlation between **average\_ride** time of each **day\_of\_week** and the **number\_of\_trips** grouped by **day\_of\_week**.

 So, we explore a little more into data, looking  which the longest trips are (in _Hours_ because of the magnitude of longest ride\_length):

In [10]:
SELECT TOP 50
	member_casual,
	started_at,
	ended_at,
	DATEDIFF(HOUR,started_at,ended_at) AS ride_length,
	DATEPART(DW, started_at) AS day_of_week,    -- DAY OF THE WEEK (1 TO 7)
	rideable_type

FROM [202103-divvy-tripdata]
ORDER BY ride_length desc


(50 rows affected)

Total execution time: 00:00:00.080

member_casual started_at ended_at ride_length day_of_week rideable_type casual 2021-03-08 16:48:24.0000000 2021-03-30 16:50:03.0000000 528 2 docked_bike casual 2021-03-19 14:11:08.0000000 2021-04-02 12:45:14.0000000 334 6 docked_bike casual 2021-03-02 17:48:41.0000000 2021-03-15 09:39:25.0000000 304 3 docked_bike casual 2021-03-06 22:37:58.0000000 2021-03-18 22:08:43.0000000 288 7 docked_bike casual 2021-03-06 22:58:09.0000000 2021-03-18 22:09:49.0000000 288 7 docked_bike casual 2021-03-22 13:22:48.0000000 2021-04-02 14:27:23.0000000 265 2 docked_bike casual 2021-03-20 17:35:12.0000000 2021-03-30 13:56:23.0000000 236 7 docked_bike casual 2021-03-18 21:23:58.0000000 2021-03-27 22:05:30.0000000 217 5 docked_bike casual 2021-03-20 23:58:11.0000000 2021-03-29 17:01:59.0000000 210 7 docked_bike casual 2021-03-21 18:20:50.0000000 2021-03-29 21:30:13.0000000 195 1 docked_bike casual 2021-03-04 06:29:39.0000000 2021-03-12 08:11:12.0000000 194 5 docked_bike casual 2021-03-27 18:45:45.0000000 2021-04-04 16:47:58.0000000 190 7 docked_bike casual 2021-03-27 18:46:55.0000000 2021-04-04 16:47:51.0000000 190 7 docked_bike casual 2021-03-29 17:51:38.0000000 2021-04-06 11:00:11.0000000 186 2 docked_bike casual 2021-03-13 18:06:47.0000000 2021-03-20 13:39:11.0000000 163 7 docked_bike casual 2021-03-13 16:11:22.0000000 2021-03-20 10:40:50.0000000 162 7 docked_bike casual 2021-03-21 03:01:27.0000000 2021-03-26 16:25:50.0000000 133 1 docked_bike casual 2021-03-27 13:06:02.0000000 2021-04-01 14:24:53.0000000 121 7 docked_bike casual 2021-03-09 21:50:04.0000000 2021-03-14 15:50:31.0000000 114 3 docked_bike casual 2021-03-08 15:21:19.0000000 2021-03-13 08:47:35.0000000 113 2 docked_bike casual 2021-03-20 23:21:49.0000000 2021-03-25 14:24:40.0000000 111 7 docked_bike casual 2021-03-29 01:46:36.0000000 2021-04-02 14:48:35.0000000 109 2 docked_bike casual 2021-03-25 10:57:10.0000000 2021-03-29 21:03:00.0000000 107 5 docked_bike casual 2021-03-13 12:41:53.0000000 2021-03-17 14:09:09.0000000 98 7 docked_bike casual 2021-03-06 15:11:04.0000000 2021-03-10 17:56:26.0000000 98 7 docked_bike casual 2021-03-30 17:56:50.0000000 2021-04-03 18:05:14.0000000 97 3 docked_bike casual 2021-03-20 16:06:03.0000000 2021-03-24 15:34:59.0000000 95 7 docked_bike casual 2021-03-30 17:56:35.0000000 2021-04-03 16:35:09.0000000 95 3 docked_bike casual 2021-03-06 12:15:04.0000000 2021-03-09 18:19:53.0000000 78 7 docked_bike casual 2021-03-27 16:08:12.0000000 2021-03-30 20:01:08.0000000 76 7 docked_bike casual 2021-03-30 16:50:55.0000000 2021-04-02 19:30:07.0000000 75 3 docked_bike casual 2021-03-24 13:55:04.0000000 2021-03-27 15:56:33.0000000 74 4 docked_bike casual 2021-03-13 11:08:25.0000000 2021-03-16 12:14:52.0000000 73 7 docked_bike casual 2021-03-10 16:54:19.0000000 2021-03-13 13:47:50.0000000 69 4 docked_bike casual 2021-03-22 17:23:38.0000000 2021-03-25 13:56:59.0000000 68 2 docked_bike casual 2021-03-06 17:19:07.0000000 2021-03-09 13:10:54.0000000 68 7 docked_bike casual 2021-03-06 22:49:31.0000000 2021-03-09 17:05:49.0000000 67 7 docked_bike casual 2021-03-14 20:41:23.0000000 2021-03-17 12:17:39.0000000 64 1 docked_bike casual 2021-03-28 01:35:15.0000000 2021-03-30 16:19:18.0000000 63 1 docked_bike casual 2021-03-29 01:38:19.0000000 2021-03-31 15:31:40.0000000 62 2 docked_bike casual 2021-03-29 23:31:08.0000000 2021-04-01 09:19:18.0000000 58 2 docked_bike casual 2021-03-06 08:39:01.0000000 2021-03-08 17:24:54.0000000 57 7 docked_bike casual 2021-03-17 10:41:28.0000000 2021-03-19 17:39:56.0000000 55 4 docked_bike casual 2021-03-07 10:49:04.0000000 2021-03-09 15:45:46.0000000 53 1 docked_bike casual 2021-03-24 13:51:00.0000000 2021-03-26 16:24:15.0000000 51 4 docked_bike casual 2021-03-22 14:46:42.0000000 2021-03-24 15:15:09.0000000 49 2 docked_bike casual 2021-03-30 16:19:11.0000000 2021-04-01 17:36:09.0000000 49 3 docked_bike casual 2021-03-06 08:10:34.0000000 2021-03-08 09:05:40.0000000 49 7 docked_bike casual 2021-03-30 13:44:27.0000000 2021-04-01 13:25:13.0000000 48

The results show an anomaly in the r**ide\_length** of the longest rides. Some insights we capture:

- The duration of the longest ride is 528 hours, which is too much in my opinion. 
- Then the second one is 334 hours, and so on until completing the top 100 with 25 hours. 
- All the three-digits duration rides are which **rideable\_type** are _'docked\_bikes'_ and the type of rider (**member\_casual**) is _casual ._
- We also found that these very long rides are mainly from _casual riders_. 
- The longest ride from _member riders_ is 26 hours (row n°80).

Now we take a deeper look at  **rideable\_types** separated by types of users (**member\_casual**):

In [3]:
SELECT  
	member_casual,
	rideable_type,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		member_casual, 
		rideable_type,
		DATEDIFF(MINUTE,started_at, ended_at) AS ride_length,  --CREAR UNA TABLA TABLA CON TODOS LOS CALCULOS 
		DATEPART(DW, started_at) AS day_of_week	
	FROM [202103-divvy-tripdata]
) AS temp_table
WHERE member_casual='casual'
GROUP BY member_casual, rideable_type
ORDER BY member_casual, number_of_trips DESC


SELECT  
	member_casual,
	rideable_type,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		member_casual, 
		rideable_type,
		DATEDIFF(MINUTE,started_at, ended_at) AS ride_length,  --CREAR UNA TABLA TABLA CON TODOS LOS CALCULOS 
		DATEPART(DW, started_at) AS day_of_week	
	FROM [202103-divvy-tripdata]
) AS temp_table
WHERE member_casual='member'
GROUP BY member_casual, rideable_type
ORDER BY member_casual, number_of_trips DESC


(3 rows affected)

(2 rows affected)

Total execution time: 00:00:00.453

member_casual rideable_type number_of_trips average_ride total_time_trips casual classic_bike 45528 31 1435951 casual electric_bike 22848 21 492410 casual docked_bike 15657 81 1278390

member_casual rideable_type number_of_trips average_ride total_time_trips member classic_bike 107017 14 1519896 member electric_bike 37446 13 498135

We can see that docked\_bike rides  have an **average\_ride** time of 81 minutes wich explain what we mentioned above. It would be useful know which are the trips that is most repeated (or Mode), grouped by **ride\_legth** time. For that we have to round the ride\_length variable removing decimals to simplify the results table:

In [8]:
SELECT  TOP 50
	ride_length,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		ROUND(DATEDIFF(MINUTE,started_at, ended_at),0) AS ride_length,  -- USE ROUND FUNCTIONS TO REMOVE DECIMALS
		DATEPART(DW, started_at) AS day_of_week	
	FROM [202103-divvy-tripdata]
) AS temp_table
GROUP BY ride_length
ORDER BY number_of_trips DESC

(50 rows affected)

Total execution time: 00:00:00.054

ride_length number_of_trips average_ride total_time_trips 6 12578 6 75468 7 12442 7 87094 5 12337 5 61685 8 11821 8 94568 4 11168 4 44672 9 10857 9 97713 10 10043 10 100430 11 9116 11 100276 12 8531 12 102372 3 8199 3 24597 13 7755 13 100815 14 7147 14 100058 15 6393 15 95895 16 5867 16 93872 17 5359 17 91103 18 4934 18 88812 2 4564 2 9128 19 4539 19 86241 20 4208 20 84160 21 3788 21 79548 22 3571 22 78562 23 3350 23 77050 24 3174 24 76176 25 3059 25 76475 26 2924 26 76024 27 2657 27 71739 28 2439 28 68292 29 2311 29 67019 1 2304 1 2304 30 2264 30 67920 31 2024 31 62744 32 1893 32 60576 0 1782 0 0 33 1684 33 55572 34 1580 34 53720 35 1492 35 52220 36 1375 36 49500 37 1238 37 45806 38 1222 38 46436 39 1149 39 44811 40 1045 40 41800 41 1011 41 41451 43 892 43 38356 42 892 42 37464 45 800 45 36000 44 723 44 31812 46 646 46 29716 47 605 47 28435 49 577 49 28273 48 534 48 25632

The mode of the variable **ride\_length** is 6 minutes followed by 7, 5, 8, 4, 9 minutes and so on (rounding the numbers to whole number closer, removing decimals). As you noticed, the **ride\_length** and **average\_ride** are equals because what we explained. Fo r example: the **average\_ride** time grouped by **ride\_length** of 6 minutes, is calculated by numbers between 5.5 and 6.4 (5.5,6.5\], so it is logical that the result tends towards the nearest whole number, in this case 6 minutes.

**<u>Removing Outliers</u>**

Now we try to find which cells we have to remove in order to achieve statistical significance. At 95% of confidence level we need z-core values lower than -1.96 or higher than 1.96. For 99% of confidence level we include the values with z-score lower than -2.96 and higher than 2.96.

In [14]:
WITH trips1 AS (
	SELECT ride_id,
	member_casual,
	ROUND(DATEDIFF(MINUTE, started_at, ended_at),0) AS ride_length
	FROM [202103-divvy-tripdata]
	),
	rides_stats AS (
	SELECT
		AVG(ride_length) AS ride_average,
		STDEV(ride_length)	AS stan_dev
	FROM (SELECT member_casual, DATEDIFF(MINUTE,started_at, ended_at) AS ride_length FROM [202103-divvy-tripdata]) AS temp_table
	)
SELECT TOP 500
	ride_id,
	ride_length,
	member_casual,
	ABS(trips1.ride_length - rides_stats.ride_average)/rides_stats.stan_dev AS z_score_ride
FROM trips1, rides_stats
ORDER BY z_score_ride DESC

(500 rows affected)

Total execution time: 00:00:00.168

ride_id ride_length member_casual z_score_ride 271D9E93A68C9A17 31682 casual 205,02022110947186 9F66323DE1CE4C08 20074 casual 129,8504571600483 F173092104F0C6AA 18231 casual 117,91576772528026 5752EBD1DB94ACB3 17251 casual 111,56959537255499 EBD8B407D4DE1469 17231 casual 111,4400816510708 61486C9900757ABE 15905 casual 102,85332191666903 7D3DF23CE224F140 14181 casual 91,68923912473191 79A5C8E0CB0D6C52 13002 casual 84,05440524323895 83F37001BA988B9F 12543 casual 81,0820653351768 A8A6F0B128472D2C 11710 casual 75,6878188353603 2CE0E235CD608CD6 11622 casual 75,11795846082987 B565F3DB8A064108 11402 casual 73,69330752450378 CC6D7C07DF55006B 11401 casual 73,68683183842957 766210D0152484C2 11109 casual 71,7959315047604 FE8E0FFFFBE57FFE 9813 casual 63,40344235258493 4CE44E9330B70D6F 9749 casual 62,98899844383553 C26113EB6F15B90F 8004 casual 51,68892624434 8132866080BCB368 7278 casual 46,987578154463925 CCC052F50EDF3908 6840 casual 44,15122765396018 53583F66F8F545C9 6806 casual 43,93105432743705 A5598413C29F4451 6663 casual 43,0050312188251 119CB691B8010131 6542 casual 42,221473203845754 2A4B5E818D855F78 6366 casual 41,08175245478488 4655F5AEEC5CEBFF 5925 casual 38,22597489605851 083BA3DB4E497CED 5848 casual 37,72734706834438 BD9726A4C58D43EF 5769 casual 37,215767868481834 68AF0FB79B7DC8DF 5728 casual 36,95026473943924 D5BF31A5EB47EDD3 5679 casual 36,63295612180298 C99F7D674E6A880A 4684 casual 30,189648477964557 B3FBA91BAAC59C53 4553 casual 29,341333602243115 CEA2873C2ED88D6E 4480 casual 28,868608518825823 84964A7EF3110B69 4441 casual 28,616056761931656 9CED6F8873289684 4386 casual 28,259894027850134 C09FB7BE559B2740 4133 casual 26,621545451075136 7297840D28F880C7 4113 casual 26,49203172959095 B8E4D47290A5AB74 4071 casual 26,22005291447415 A5106DA7FD8B59AE 3976 casual 25,60486273742425 A63F673040F66730 3816 casual 24,568752965550736 F4FFEA4C065B79DB 3764 casual 24,232017289691843 44F8675161BF37CA 3713 casual 23,90175729990716 6CE4D2B17664782A 3468 casual 22,315214211725838 9D7CCC419E706275 3405 casual 21,90724598905064 05FF29F0587850B5 3298 casual 21,21434757911023 DA6006141527639C 3176 casual 20,424313878056672 2984FBF69A3A4846 3033 casual 19,49829076944472 211BCD29059D2E19 2957 casual 19,0061386278048 286E710824F74EBC 2935 casual 18,86367353417219 D9E885C7DEA44F9E 2909 casual 18,695305696242745 3C2F8A9877F15942 2861 casual 18,38447276468069 37AAD5EE100AE6E4 2774 casual 17,821088076224466 CC13B7628EE4DB98 2774 casual 17,821088076224466 7D3B9F82CFFEDB8A 2756 casual 17,704525726888694 33EDF65F6016830E 2698 casual 17,328935934584546 AAEF4910E9A68AF8 2683 casual 17,231800643471402 4AE86C0D4A537E00 2653 casual 17,037530061245118 DF5161DC17B8D59B 2641 casual 16,959821828354606 8EC9777B2D17D05F 2613 casual 16,77850261827674 0692AD88F066D77B 2589 casual 16,623086152495713 1DB0ED8F508E0630 2569 casual 16,493572431011525 559C7DBC3BA00F71 2542 casual 16,318728907007866 FED267F132C2866B 2535 casual 16,2733991044884 C0FC0B1BD299D2E8 2462 casual 15,80067402107111 E2787EC004B28D86 2461 casual 15,794198334996901 C8849E43E2E061C5 2461 casual 15,794198334996901 C518D8B22D43D920 2449 casual 15,716490102106388 B6DB9FFE7ECBF9DC 2407 casual 15,44451128698959 52B003B9C6FC31E8 2368 casual 15,19195953009542 71A48DBF01C043BF 2289 casual 14,680380330232872 AC9867BAA0E24E5F 2199 casual 14,09756858355402 DD5B8BE0217DF562 1893 casual 12,116008644845921 AC81B43066844207 1812 casual 11,591478072834954 F498C2FB5BA2402C 1811 casual 11,585002386760744 87F53011E84BD82A 1702 casual 10,879152604671912 32EF2C2B64B09813 1688 casual 10,788492999632979 DD8E1A54C4A18131 1674 casual 10,697833394594047 29951372E15D1F57 1659 casual 10,600698103480905 BB93FF421B81947E 1658 casual 10,594222417406696 77355DA3F172D759 1648 casual 10,5294655566646 333D70AFDE4D1EDB 1590 casual 10,153875764360452 B3299E0D9E3E687D 1560 casual 9,959605182134167 FF3F8161C12BAC5D 1560 casual 9,959605182134167 BACE0C8FC3376FB4 1560 member 9,959605182134167 0316B5F9AD0AD991 1560 casual 9,959605182134167 70B8A78

The results showed that values with **ride\_length** time greater than 480 minutes (row n°433) have a z-score value higher than 2.96 (absolute value). This give us a confidence level of 95%. 

If we exclude all theese data from the dataset, we will see how the average

In [18]:
--AISLANDO LOS OUTLIERS (las docked_bikes arrojaban valores ride_length muy elevados)

SELECT  
	member_casual,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		member_casual, 
		DATEDIFF(MINUTE,started_at, ended_at) AS ride_length  --CREAR UNA TABLA TABLA CON TODOS LOS CALCULOS 
	FROM [202103-divvy-tripdata]
) AS temp_table
GROUP BY member_casual

-- REMOVING OUTLIERS > 430 MINUTES
SELECT  
	member_casual,
	COUNT(ride_length) AS number_of_trips, 
	AVG(ride_length) AS average_ride, 
	SUM(ride_length) AS total_time_trips

FROM (
	SELECT	
		member_casual, 
		DATEDIFF(MINUTE,started_at, ended_at) AS ride_length  --CREAR UNA TABLA TABLA CON TODOS LOS CALCULOS 
	FROM [202103-divvy-tripdata]
) AS temp_table
WHERE ride_length < 480 
GROUP BY member_casual      

(2 rows affected)

(2 rows affected)

Total execution time: 00:00:00.112

member_casual number_of_trips average_ride total_time_trips casual 84033 38 3206751 member 144463 13 2018031

member_casual number_of_trips average_ride total_time_trips casual 83675 29 2438181 member 144415 13 1967721